hyperbolic-rs/preprocess.py

```diff
@@ -17,30 +17,50 @@
 import tensorflow as tf
 import numpy as np
 import random
-from scipy import sparse
+from tqdm import tqdm
 from pathlib import Path
-from rudders.datasets import movielens, keen
+from rudders.relations import Relations
+from rudders.datasets import movielens, keen, amazon
 from rudders.config import CONFIG
 from rudders.utils import set_seed, sort_items_by_popularity, save_as_pickle
 
 FLAGS = flags.FLAGS
-flags.DEFINE_string('run_id', default='prep-notitle-hopdist0.75', help='Name of prep to store')
-flags.DEFINE_string('item', default='ml-1m', help='Item can be "keen" (user-keen interactions), "gem" (keen-gem '
-                                                'interactions), or "ml-1m"')
-flags.DEFINE_string('dataset_path', default='data/ml-1m', help='Path to raw dataset: data/keen, data/ml-1m')
-flags.DEFINE_string('item_item_file', default='data/prep/ml-1m/item_item_notitle_hop_distance_th0.75.pickle',
+flags.DEFINE_string('run_id', default='musicins-top10', help='Name of prep to store')
+flags.DEFINE_string('item', default='amzn-musicins', help='Item can be "keen" (user-keen interactions), '
+                                                          '"gem" (keen-gem interactions), "ml-1m", '
+                                                          '"amzn-musicins", "amzn-vgames"')
+flags.DEFINE_string('dataset_path', default='data/amazon', help='Path to raw dataset: data/keen, data/ml-1m, '
+                                                               'data/amazon')
+flags.DEFINE_string('item_item_file', default='data/prep/amazon/musicins_musicins_cosine_distance_th0.6.pickle',
                     help='Path to the item-item distance file')
 flags.DEFINE_boolean('plot_graph', default=False, help='Plots the user-item graph')
-flags.DEFINE_boolean('shuffle', default=False, help='Shuffle the samples')
-flags.DEFINE_boolean('sparse', default=False, help='Stores item-item matrix as a sparse matrix')
+flags.DEFINE_boolean('shuffle', default=False, help='Whether to shuffle the interactions or not')
+flags.DEFINE_boolean('add_extra_relations', default=True, help='For the amazon dataset, adds co-buy, co-view and'
+                                                                'categorical similarity relations')
 flags.DEFINE_integer('min_user_interactions', default=5, help='Users with less than this interactions are filtered')
 flags.DEFINE_integer('min_item_interactions', default=2, help='Items with less than this interactions are filtered')
 flags.DEFINE_integer('max_item_interactions', default=150, help='Items with more than this interactions are filtered')
+flags.DEFINE_integer('similarity_items_per_item', default=10, help='Amount of similarity items to add per item')
```

Comment:
Would be nice to see an overview which of these you treated as hyper-parameters.

Reply:
In this case, I tried with 10 and 25. 10 gave better results so I kept it.